Presentation.Web/app/components/data-processing/tabs/data-processing-registration-sub-data-processor-modal.controller.ts

```diff
@@ -0,0 +1,189 @@
+        subDataProcessorInsecureThirdCountriesConfig: Models.ViewModel.Generic.ISingleSelectionWithFixedOptionsViewModel<Models.Generic.NamedEntity.NamedEntityWithDescriptionAndExpirationStatusDTO>;
+
+        title: string;
+        subDataProcessorFieldName: string;
+        isEdit: boolean;
+        subDataProcessorName: string;
+
+        viewModel: Models.ViewModel.GDPR.SubDataProcessorViewModel;
+
+        readonly yesValue = Models.Api.Shared.YesNoUndecidedOption.Yes;
+
+        private isBusy = false;
+
+        constructor(
+            private readonly subDataProcessorId: number | null,
+            private readonly dataProcessingRegistration: Models.DataProcessing.IDataProcessingRegistrationDTO,
+            private readonly dataProcessingRegistrationOptions: Models.DataProcessing.IDataProcessingRegistrationOptions,
+            private readonly $uibModalInstance,
+            private readonly select2LoadingService: Services.ISelect2LoadingService,
+            private readonly dataProcessingRegistrationService: Services.DataProcessing.IDataProcessingRegistrationService) {
+        }
+
+        $onInit() {
+            
+            const subDataProcessor = this.getSubDataProcessor();
+
+            this.isEdit = !!subDataProcessor;
+            this.subDataProcessorName = this.isEdit ? subDataProcessor.name : "";
+
+            const titleSuffix = "underdatabehandler";
+            this.subDataProcessorFieldName = "Underdatabehandler";
+            this.title = subDataProcessor ? `Rediger ${titleSuffix}` : `Opret ${titleSuffix}`;
+
+            this.viewModel = new Models.ViewModel.GDPR.SubDataProcessorViewModel(this.subDataProcessorId,
+                subDataProcessor?.cvrNumber,
+                subDataProcessor?.basisForTransfer?.id,
+                subDataProcessor?.transferToInsecureThirdCountries,
+                subDataProcessor?.insecureCountry?.id);
+
+            this.subDataProcessorsConfig = this.bindSubDataProcessor();
+            this.subDataProcessorBasisForTransferConfig = this.bindBasisForTransfer(subDataProcessor);
+            this.subDataProcessorTransferToThirdCountriesConfig = this.bindTransferToThirdCountriesOptions(subDataProcessor);
+            this.subDataProcessorInsecureThirdCountriesConfig = this.bindInsecureThirdCountries(subDataProcessor);
+        }
+
+        save(): void {
+            if (this.isBusy)
+                return;
+            this.isBusy = true;
+
+            const payload = this.viewModel.prepareRequestPayload();
+            if (this.subDataProcessorId) {
+                this.dataProcessingRegistrationService
+                    .updateSubDataProcessor(this.dataProcessingRegistration.id, payload)
+                    .then(() => {
+                            this.isBusy = false;
+                            this.cancel(true);
```

Comment:
looks wierd that it calls "cancel" since it actually assigned something... consider renaming the "cancel" or actually making a "close" which takes a "result" that it passes back